### PR TITLE
WIP - Detect if running on a Mac M1 and use appropriate dev-requirements file

### PR DESCRIPTION
**Pull request by eapolinario**
Signed-off-by: Eduardo Apolinario <eapolinario@users.noreply.github.com>

Fixes https://github.com/flyteorg/flyte/issues/3264

# TL;DR
Special case Mac M1/2's in the case of dev environment  

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
In this PR the make targets related to the dev environment detect if the user's platform. We introduce a new dev-requirements file produced on a Mac M1.

Testing:

- [ ] need someone to run `make dev-requirements.txt` on a M1 and push a commit to this PR with the contents of the `dev-requirements-mac_arm64.txt` file.
- [ ] run `make setup` on a new venv

## Tracking Issue
https://github.com/flyteorg/flyte/issues/3264

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by eapolinario** [comment]
Superseded by https://github.com/flyteorg/flytekit/pull/1511.
